cli/src/commands/rebase.rs

```diff
@@ -344,10 +358,22 @@ fn rebase_revision(
         .iter()
         .commits(workspace_command.repo().store())
         .try_collect()?;
-    // Currently, immutable commits are defied so that a child of a rewriteable
+    // Currently, immutable commits are defined so that a child of a rewriteable
     // commit is always rewriteable.
     debug_assert!(workspace_command.check_rewritable(&child_commits).is_ok());
 
+    // Skip rebasing if `old_commit` is already a child of `new_parents`
+    // and has no children to rebase on top of `old_commit`, to avoid creating a new
+    // transaction.
+    if old_commit.parents().eq(&new_parents) && child_commits.is_empty() {
+        writeln!(
+            ui.stderr(),
+            "Skipping rebase of commit {}",
+            short_commit_hash(old_commit.id()),
+        )?;
+        return Ok(());
+    }
```

Comment:
Perhaps, this is redundant because we'll check `new_parents` anyway after noop rebase of descendants.